### PR TITLE
fix: make-installer possible break typings

### PR DESCRIPTION
**Pull request by exherb**
ref https://github.com/element-plus/element-plus/pull/8106 / https://github.com/element-plus/element-plus/issues/8122

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by jw-foss** [comment]
@sxzz Please take a look on this issue

**Comment by tolking** [comment]
@exherb Please check type

**Comment by sxzz** [comment]
image

Can't work

**Comment by sxzz** [comment]
I think the problem is fixed in #8106.

**Comment by exherb** [comment]
> #8106

I don't think this is a fix if some libs / project use @vue/runtime-core.

I will take a look about this typing issues.

**Comment by exherb** [comment]
I have make some changes, please review again when got time @sxzz

File: packages/element-plus/make-installer.ts [hunk 1]
```diff
@@ -5,17 +5,16 @@ import type { ConfigProviderContext } from '@element-plus/tokens'
...
-declare module '@vue/runtime-core' {
-  interface App {
-    [INSTALLED_KEY]?: boolean
-  }
+interface ElementPlusApp extends App {
+  [INSTALLED_KEY]: boolean
 }
 
 export const makeInstaller = (components: Plugin[] = []) => {
   const install = (app: App, options?: ConfigProviderContext) => {
```
Comment (sxzz) [inline packages/element-plus/make-installer.ts:13]: Forgot about it.
Just change it to `any`. I think the type of `app` is not important.
```suggestion
  const install = (app: any, options?: ConfigProviderContext) => {
```

**Comment by tolking** [comment]
> > #8106
> 
> I don't think this is a fix if some lib / project import `App` from @vue/runtime-core instead of vue
> 
> I will take a look about the pasted typing issues.

Are you sure it doesn't work?

**Comment by exherb** [comment]
> > > #8106
> > 
> > 
> > I don't think this is a fix if some lib / project import `App` from @vue/runtime-core instead of vue
> > I will take a look about the pasted typing issues.
> 
> Are you sure it doesn't work?

It is work in some scenarios but not all scenarios.

**Comment by tolking** [comment]
@exherb show me the code. I just want to make sure how to modify it.

**Comment by tolking** [comment]
@exherb show me the code. I just want to make sure how to modify it.

**Comment by exherb** [comment]
https://github.com/element-plus/element-plus/pull/8130

At least reference it to acknowledge, I just want to make contribute. By doing this, I feel my contribution is not welcome here. @jw-foss

**Comment by jw-foss** [comment]
@exherb I think this was a mistake, and everybody is welcome to contribute that is for sure. In discussions like these, there were gaps between outside collaborators (not in a bad way), would you mind joining our discord platform so that we can discuss more conveniently and more quickly? Here is the [link](https://discord.link/ElementPlus)

**Comment by sxzz** [comment]
@exherb Sorry about this PR. IMHO because there's another way to solve this issue. I just wanted to explore another possibility. This does not mean that this PR will be rejected by us. Furthermore your PR is unable to solve the issue (when I create the other PR), so I just want to solve it, that's it. We can still discuss it if you would like to. 
Hope you can understand my behavior better and you and your PR are not unwelcome.

**Comment by ac51300696** [comment]
`
import { provideGlobalConfig } from '@element-plus/hooks'
import { version } from './version'
import type { App, Plugin } from 'vue'
import type { ConfigProviderContext } from '@element-plus/tokens'

const INSTALLED_KEY = Symbol('INSTALLED_KEY')

interface elementApp {
      [INSTALLED_KEY]?: boolean
    }

export const makeInstaller = (components: Plugin[] = []) => {
  const install = (app: App & elementApp, options?: ConfigProviderContext) => {
    if (app[INSTALLED_KEY]) return

    app[INSTALLED_KEY] = true
    components.forEach((c) => app.use(c))

    if (options) provideGlobalConfig(options, app, true)
  }

  return {
    version,
    install,
  }
}

`
how about that?

**Pull request closed without merging**